### PR TITLE
Feature/Any spark source and sink

### PR DESCRIPTION
**Pull request by hayssams**
## Summary
Supports JDBC / Amazon Redshift / Snowflake source or sink database for input files and/or any spark source/sink including Amazon Redshift
**Related Issue: #IssueId**

**PR Type: Feature**

**Status: Ready to review**

**Breaking change? No**

### CODE REVIEW

File: src/main/scala/com/ebiznext/comet/config/Settings.scala [hunk 1]
```diff
@@ -111,13 +111,13 @@ object Settings extends StrictLogging {
     *       (e.g. non-standard table names) alongside with the regular schema definition, on the same
```
Comment (mhdkassir) [inline src/main/scala/com/ebiznext/comet/config/Settings.scala:111]: scaladoc needs to be updated, format, mode, and options replace uri,user, and password

File: src/main/resources/reference.conf [hunk 2]
```diff
@@ -137,24 +141,22 @@ jdbc-engines {
     driver = "org.h2.Driver"
```
Comment (mhdkassir) [inline src/main/resources/reference.conf:141]: `driver` can now be removed from `JdbcEngine` it is not used anymore. It is now set in the `options` of `Connection`

File: src/main/scala/com/ebiznext/comet/job/index/connectionload/ConnectionLoadJob.scala [hunk 3]
```diff
@@ -31,18 +26,20 @@ class JdbcLoadJob(
           case Left(path) => session.read.parquet(path)
           case Right(df)  => df
         }
-      sourceDF.write
-        .format("jdbc")
-        .option("numPartitions", cliConfig.partitions)
-        .option("batchsize", cliConfig.batchSize)
+
+      // Some database do not suport truncate wuring save
```
Comment (mhdkassir) [inline src/main/scala/com/ebiznext/comet/job/index/connectionload/ConnectionLoadJob.scala:30]: a small typo ;) during

File: src/main/scala/com/ebiznext/comet/config/Settings.scala (hunk 1 above)
Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/config/Settings.scala:114]: Done

File: src/main/resources/reference.conf (hunk 2 above)
Comment (hayssams) [reply src/main/resources/reference.conf]: Done

File: src/main/scala/com/ebiznext/comet/job/index/connectionload/ConnectionLoadJob.scala (hunk 3 above)
Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/job/index/connectionload/ConnectionLoadJob.scala]: Done

**Review by mhdkassir — APPROVED** [review]
LGTM

**Pull request merged**